infra/src/main/kotlin/fi/fta/geoviite/infra/split/SplitValidation.kt

```diff
@@ -76,44 +78,49 @@ internal fun validateSplitContent(
     return listOf(multipleSplitsStagedErrors, contentErrors).flatten()
 }
 
+const val MAX_SPLIT_POINT_OFFSET = 1.0
+
 internal fun validateTargetGeometry(
-    targetAddresses: AlignmentAddresses?,
-    sourceAddresses: AlignmentAddresses?,
+    operation: SplitTargetOperation,
+    targetPoints: List<AddressPoint>?,
+    sourcePoints: List<AddressPoint>?,
 ): PublicationValidationError? {
-    return if (targetAddresses == null || sourceAddresses == null) {
+    return if (targetPoints == null || sourcePoints == null) {
         PublicationValidationError(ERROR, "$VALIDATION_SPLIT.no-geometry")
     } else {
-        val targetPoints = targetAddresses.allPoints.filter { addressPoint -> addressPoint.address.hasZeroMillimeters }
-        val sourcePoints = sourceAddresses.allPoints.filter { addressPoint -> addressPoint.address.hasZeroMillimeters }
-        val startIndex = max(0, sourcePoints.indexOfFirst { it.isSame(targetPoints.first()) })
+        // Geocoding mid-points are the even meter-points, but assert just in case
+        val startIndex = max(0, sourcePoints.indexOfFirst { s -> s.address == targetPoints.first().address })
 
         targetPoints
             .withIndex()
-            .firstOrNull { (targetIndex, targetPoint) ->
-                val idx = min(sourcePoints.lastIndex, startIndex + targetIndex)
-                !targetPoint.isSame(sourcePoints[idx])
-            }?.let { (_, point) ->
-                PublicationValidationError(
-                    ERROR,
-                    "$VALIDATION_SPLIT.geometry-changed",
-                    mapOf("point" to point),
-                )
+            .firstNotNullOfOrNull { (targetIndex, targetPoint) ->
+                val sourcePoint = sourcePoints.getOrNull(startIndex + targetIndex)
+                if (sourcePoint?.address != targetPoint.address) {
+                    PublicationValidationError(ERROR, "$VALIDATION_SPLIT.trackmeters-changed")
+                } else if (operation != SplitTargetOperation.TRANSFER && !targetPoint.point.isSame(sourcePoint.point)) {
```

Comment:
Muuttaisin ehkä nämä `else if`:it`when`-rakenteeksi jossa käydään läpi `SplitTargetOperation`:it. Jos meille koskaan tulee lisää operaatiotyypejä (vaikkakin se on näillä näkymin melko epätodennäköistä) niin tämä ehto ei välttämättä enää päde, ja silloin on paljon helpompaa huomata että tätä koodia pitää muuttaa jos/kun kääntäjäkin siitä valittaa

Reply:
Njoo totta ja ei tuo kyllä muutenkaan kovin kaunis ole. Toisaalta when tarkoittaisi väkisin syvempää rakennetta koska 1. if on täysin tyypistä riippumaton... Voisin ehkä erottaa tuon operaation ja piste-samuuden tarkastelun omaksi funkkariksi, jolloin sen voisi tehdä turvallisemmin.

Reply:
äh... ei sekään mene kovin nätisti, koska enumissa on 2 arvoa joilla katsotaan isSame ja whenissäkin tuo kannattaisi hoitaa elsessä. Lisäksi noista tulee eri virheet tarkoituksella. Ei tuo kaunis ole mutta jätän toistaiseksi näin kun en oikein keksi nätimpääkään.